components/brave_wallet/browser/tx_storage_delegate_impl.h

```diff
@@ -0,0 +1,72 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_TX_STORAGE_DELEGATE_IMPL_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_TX_STORAGE_DELEGATE_IMPL_H_
+
+#include <memory>
+
+#include "base/gtest_prod_util.h"
+#include "base/memory/scoped_refptr.h"
+#include "base/memory/weak_ptr.h"
+#include "base/observer_list.h"
+#include "base/task/sequenced_task_runner.h"
+#include "brave/components/brave_wallet/browser/tx_storage_delegate.h"
+#include "components/value_store/value_store_factory.h"
+
+class PrefService;
+
+namespace value_store {
+class ValueStoreFrontend;
+}  // namespace value_store
+
+namespace brave_wallet {
+
+class TxStorageDelegateImpl final : public TxStorageDelegate {
+ public:
+  TxStorageDelegateImpl(
+      PrefService* prefs,
+      scoped_refptr<value_store::ValueStoreFactory> store_factory,
+      scoped_refptr<base::SequencedTaskRunner> ui_task_runner);
+  ~TxStorageDelegateImpl();
+  TxStorageDelegateImpl(const TxStorageDelegateImpl&) = delete;
+  TxStorageDelegateImpl& operator=(const TxStorageDelegateImpl&) = delete;
+
+  bool IsInitialized() const override;
+  const base::Value::Dict& GetTxs() const override;
+  base::Value::Dict& GetTxs() override;
+  void ScheduleWrite() override;
```

Comment:
nit: There could be similar to PrefService's ScopedPrefUpdate utility class which allows `GetTxs()` by non-const ref and does `ScheduleWrite()` in dtor

Reply:
fixed in 330c37ab4a9a518d77a58769424ffe8b45e84776